cpp/src/community/detail/mis_impl.cuh

```diff
@@ -0,0 +1,793 @@
+                         }
+                       }
+                     });
+
+    if (debug) {
+      CUDA_TRY(cudaDeviceSynchronize());
+      std::cout << "Rank of maximum rank neighbors" << std::endl;
+      thrust::for_each(
+        handle.get_thrust_policy(),
+        max_outgoing_ranks.begin(),
+        max_outgoing_ranks.end(),
+        [] __device__(auto max_rank_neighbor) { printf("%d \n", max_rank_neighbor); });
+    }
+
+    if (debug) std::cout << "----------------count-------------------" << std::endl;
+    vertex_t nr_candidates_to_remove = thrust::count_if(
+      handle.get_thrust_policy(),
+      remaining_vertices.end() - nr_candidates,
+      remaining_vertices.end(),
+      [max_rank_neighbor_first = max_outgoing_ranks.begin(),
+       temporary_ranks =
+         raft::device_span<vertex_t const>(temporary_ranks.data(), temporary_ranks.size()),
+       ranks   = raft::device_span<vertex_t>(ranks.data(), ranks.size()),
+       v_first = graph_view.local_vertex_partition_range_first(),
+       debug   = debug] __device__(auto v) {
+        auto v_offset          = v - v_first;
+        auto max_neighbor_rank = *(max_rank_neighbor_first + v_offset);
+
+        auto rank_of_v     = ranks[v_offset];
+        auto tmp_rank_of_v = temporary_ranks[v_offset];  // debug
+
+        if (debug) {
+          bool valid = (max_neighbor_rank < std::numeric_limits<vertex_t>::max()) &&
+                       (max_neighbor_rank > invalid_vertex_id<vertex_t>::value);
+          printf("%d, (r= %d, t= %d)  ==> %d [t= %d, r= %d]\n",
+                 v,
+                 rank_of_v,
+                 tmp_rank_of_v,
+                 max_neighbor_rank,
+                 valid ? temporary_ranks[max_neighbor_rank - v_first] : max_neighbor_rank,
+                 valid ? ranks[max_neighbor_rank - v_first] : max_neighbor_rank);
+        }
+
+        if (max_neighbor_rank >= std::numeric_limits<vertex_t>::max()) {
+          if (debug) { printf("---> to discard %d\n", v); }
+
+          // Maximum rank neighbor is alreay in MIS
+          // Discard current vertex by setting (global) rank to -Inf
+          return true;
+        }
+
+        if (rank_of_v > max_neighbor_rank) {
+          if (debug) { printf("---> to include %d\n", v); }
+          // Mark it included by setting (global) rank to +Inf
+          return true;
+        }
+        if (debug) { printf("\n"); }
+        return false;
+      });
+
+    std::cout << "check out degree of candidates\n";
+
+    thrust::for_each(handle.get_thrust_policy(),
+                     remaining_vertices.end() - nr_candidates,
+                     remaining_vertices.end(),
+                     [out_degrees = out_degrees.data(),
+                      v_first     = graph_view.local_vertex_partition_range_size()] __device__(auto v) {
+                       if (out_degrees[v - v_first] <= 0) {
+                         printf("?v=%d, out_degrees[%d] = \n", v, v, out_degrees[v - v_first]);
```

Comment:
This requires two integers but you're providing 3. It is very likely that compiler is complaining about this. Please fix warnings as soon as you see them and don't wait till code cleanup.